File: 30-days-of-code/day01/solution.go
package main

import (
	"fmt"
	"os"
	"bufio"
	"strconv"
)

func main() {
	var _ = strconv.Itoa // Ignore this comment. You can still use the package "strconv".

	var i uint64 = 4
	var d float64 = 4.0
	var s string = "HackerRank "

	scanner := bufio.NewScanner(os.Stdin)

	// Declare second integer, double, and String variables.
	var i_input uint64
	var d_input float64
	var s_input string

	// Read and save an integer, double, and String to your variables.
	scanner.Scan()
	input := scanner.Text()
	i_input, err := strconv.ParseUint(input, 10, 64)
	if err != nil {
		panic(fmt.Sprintf("Invalid int: %v", input))
	}

	scanner.Scan()
	input = scanner.Text()
	d_input, err = strconv.ParseFloat(input, 64)
	if err != nil {
		panic(fmt.Sprintf("Invalid float: %v", input))
	}

	scanner.Scan()
	s_input = scanner.Text()

	// Print the sum of both integer variables on a new line.
	fmt.Println(i + i_input)

	// Print the sum of the double variables on a new line.
	fmt.Printf("%.1f\n", d + d_input)

	// Concatenate and print the String variables on a new line
	// The 's' variable above should be printed first.
	fmt.Println(s + s_input)
}

File: 30-days-of-code/day01/solution.swift
var i = 4
var d = 4.0
var s = "HackerRank "

// This program starts with the three variables defined above:
// i, an integer; d, a double; and s, a string. We'll use those
// variables in our computation below.

// First, we read an integer from stdin, add that to the
// integer defined above, and print the result.
if let input = readLine() {
  if let input_i = Int(input) {
    print(i + input_i)
  }
}

// Next, we read a double from stdin, add that to the double
// defined above, and print the result.
if let input = readLine() {
  if let input_d = Double(input) {
    print(d + input_d)
  }
}

// And lastly, we read a string, concatenate it to the string
// defined above, and print the result.
if let input = readLine() {
  print(s + input)
}
